### DISCUSSION TITLE
Feed tft_espi with custom method for the touch controller

### DISCUSSION THREAD

**Discussion by beckmx** [Bodmer/TFT_eSPI#2339] [New processor/display support or capability request]
Hello guys, so I managed to create a custom PCB for a screen I found online (https://es.aliexpress.com/item/4001248271603.html?spm=a2g0o.order_list.order_list_main.277.5438194dlGiwCK&gatewayAdapt=glo2esp), I got it working as an ILI9488, looks good and works great, however the touch controller is something I would like to add to TFT_eSPI, I have tested with the Adafruit library for FT6206 and also works pretty great, so I want to know how could I feed tft_espi with a custom method to pass the "touched points" on screen. my main app is using LVGL so I can inject that into LVGL but there is this "welcome screen" in tft_espi the "touch calibration" that I would like to feed with this new custom method. Is it possible?

Greetings!

**Comment by Bodmer** [comment]
The calibration functions in TFT_eSPI are specific to the included XPT2046 touch controller code so will not work with other touch libraries.

Typically the map() function is used inside the sketch to map FT6206 touch coordinates to the screen.

**Comment by GeorgeIoak** [comment]
[buttonAction_UpDownCounter.zip](https://github.com/Bodmer/TFT_eSPI/files/10484650/buttonAction_UpDownCounter.zip)
I've never had to use calibration data with FT6206 CTP screens.

I've attached an example project I did to test long press events with buttons.

I think case 2 in this sketch (in `if (ts.touched())`)  is wrong and it should be:

```
case 2:
        //*x = p.x;
        t_x = tft.width() - p.x;
        t_y = tft.height() - p.y;
```

Since writing this sketch a lot has changed in my code but hopefully this example helps you integrate the CTP with TFT-eSPI
